src/components/Select2.vue

```diff
@@ -1,37 +1,18 @@
 <template>
-  <section>
-    <div ref="select" class="select" @click="() => { toggleSelect() }">
-      <div class="select-holder">{{ setedOption || '-- Select an option' }}</div>
-    </div>
-    <div class="options">
-      <div class="search">
-        <input @input="() => { search() }" type="text" v-model.trim="searchValue"/>
-      </div>
-      <div v-if="!options.length" class="info">
-        No data.
-      </div>
-      <div v-else-if="!selectOptions.length && !searchValue" class="info">
-        Loading...
-      </div>
-      <div v-else-if="!selectOptions.length && searchValue" class="info">
-        No results found.
-      </div>
-      <div v-else-if="selectOptions.length">
-        <div v-if="selectWithGroups">
-          <div v-for="group in selectOptions" :key="group.name" class="option-group">
-          <span class="option-group-name">{{ group.name }}</span>
-            <div v-for="option in group.options" @click="() => { setOption(option) }" :key="option.id" class="option">
-              {{ optionView(option) }}
-            </div>
-          </div>
-        </div>
-        <div v-else v-for="option in selectOptions" @click="() => { setOption(option) }" :key="option.id" class="option">
-          {{ optionView(option) }}
-        </div>
-      </div>
-
-    </div>
-  </section>
+  <select ref="select" class="select" @change="onSelectChange">
```

Comment:
This doesn't need to be in a `Select2.vue` component any more and can be just part of the `DataSourceProvider.vue` file. We're not creating anything unique or reusable here that requires this many levels of abstraction.

Reply:
@tonytlwu Deleted `Select2.vue` and moved its logic to the `DataSourceProvider.vue`.